src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1058,6 +1060,47 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._terminalAltBufferActiveContextKey.set(!!(this.xterm && this.xterm.raw.buffer.active === this.xterm.raw.buffer.alternate));
 	}
 
+	private async _shouldPasteText(text: string): Promise<boolean> {
+		let textForLines = text.split(/(\r\n|\n|\r)/);
+		let confirmation: IConfirmationResult;
+
+		// If the clipboard has only one line, no prompt will be triggered
+		if (textForLines.length === 1 || this._configurationService.getValue<boolean>(TerminalSettingId.MultiLinePasteWarning) === false) {
+			confirmation = { confirmed: true };
+		} else {
+			let message = nls.localize('confirmMoveTrashMessageFilesAndDirectories', "Are you sure you want to paste the following {0} lines to the terminal?", text.split(/\r\n|\n|\r/).length);
+
+			let displayItemsCount = 3;
+			let detail = textForLines.slice(0, displayItemsCount).join('');
+			if (textForLines.length > displayItemsCount) {
+				detail += '...';
+			}
```

Comment:
We could also truncate each individual line to fix the case you called out in https://github.com/microsoft/vscode/issues/122683#issuecomment-1025093731. How does a max line length of 30 work, using `…` on each line if exceeded?

Reply:
I agree, deal with https://github.com/slidoooor/vscode/blob/0bb6cac9f79af64a07126d23ca1a86b2c8363280/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1078

Reply:
> We could also truncate each individual line to fix the case you called out in [#122683 (comment)](https://github.com/microsoft/vscode/issues/122683#issuecomment-1025093731). How does a max line length of 30 work, using `…` on each line if exceeded?

I agree, deal with:
https://github.com/slidoooor/vscode/blob/0bb6cac9f79af64a07126d23ca1a86b2c8363280/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1078